src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -0,0 +1,90 @@
+6. [Initialize an Appwrite function](/docs/tooling/command-line/functions) and copy and paste your code
+
+# Develop {% #develop %}
+
+Use the `appwrite run functions` command to develop your function locally.
+
+```sh
+appwrite run functions
+? Which function would you like to develop locally? My Awesome Function ("<FUNCTION_ID>")
+
+ runtime   | entrypoint  | path                           | commands     
+-----------|-------------|--------------------------------|--------------
+ node-16.0 | src/main.js | functions/<FUNCTION_ID>      | npm install  
+
+ℹ Info: If you wish to change your local settings, update the appwrite.json file and rerun the 'appwrite run' command.
+♥ Hint: Permissions, events, CRON and timeouts dont apply when running locally.
+ℹ Info: Pulling Docker image ...
+♥ Hint: This may take a few minutes, but we only need to do this once.
+ℹ Info: Building function using Docker ...
+Preparing for build ...
+
+Building ...
+
+
+added 4 packages, and audited 5 packages in 2s
+
+
+1 package is looking for funding
+  run `npm fund` for details
+
+
+found 0 vulnerabilities
+
+Packing build ...
+
+Build finished.
+
+ℹ Info: Starting function using Docker ...
+♥ Hint: Function automatically restarts when you edit your code.
+✓ Success: Visit http://localhost:3000/ to execute your function.
+```
+
+This command helps you efficiently develop your Appwrite functions on your local machine. Modifying your function code updates your function's behavior and you can see that behavior upon execution. When developing your Appwrite function locally, it will receive [headers](/docs/products/functions/develop#headers) like when you develop a function and it is deployed to Appwrite.
+
+{% arrow_link href="/docs/products/functions/develop" %}
+Learn more about developing a function
+{% /arrow_link %}
+
+# Dynamic API keys {% #dynamic-api-keys %}
+
+You can use headers like dynamic API keys in your function, which give you access to your project services and allow you to operate without sessions. To configure your dynamic API key scopes, modify the scopes in the `appwrite.json` file.
+  
+{% arrow_link href="/docs/products/functions/develop#dynamic-api-key" %}
+Learn more about dynamic API keys
+{% /arrow_link %}
+
+# Impersonate user {% #impersonate-user %}
+
+You can also impersonate a user when you develop a function locally. Impersonate a user using the `--user-id <USER_ID>` option to select a user you want to use for testing. This allows you to test if the user can perform specific actions, such as creating a document.
+
+When using the `--user-id <USER_ID>` endpoint, the CLI will check and return an error if the user does not exist. But if a user does exist, a [JWT token](/docs/products/auth/jwt#jwt) will be generated and last for 1 hour, similar to API tokens. Additionally, the environment variable is set, allowing open runtimes to use it as a fallback instead of headers.
```

Comment:
I was referring to these Local Development release notes: https://www.notion.so/CLI-Local-functions-development-7ee3f08a99024db7ab84ffc8e22bd1f2. Otherwise I'm not sure what you mean by the headers regarding the previous comment on the impersonate a user section

Reply:
> I was referring to these Local Development release notes: https://www.notion.so/CLI-Local-functions-development-7ee3f08a99024db7ab84ffc8e22bd1f2. 

That's incorrect/outdated.

> Otherwise I'm not sure what you mean by the headers regarding the previous comment on the impersonate a user section

The `x-appwrite-user-jwt` and `x-appwrite-user-id` are set. See https://github.com/appwrite/sdk-for-cli/blob/86506a4c0a93ddcf0982bb635480bea0527d66c4/lib/commands/run.js#L163-L164